java/ingest/src/main/java/sleeper/ingest/IngestRecordsFromIterator.java

```diff
@@ -43,11 +52,18 @@ public IngestRecordsFromIterator(IngestProperties properties, Iterator<Record> r
      * @throws IteratorException    -
      */
     public long write() throws StateStoreException, IOException, InterruptedException, IteratorException {
-        IngestRecords ingestRecords = new IngestRecords(properties);
         ingestRecords.init();
         while (recordsIterator.hasNext()) {
             ingestRecords.write(recordsIterator.next());
         }
         return ingestRecords.close();
     }
+
+    public List<FileInfo> writeReturningFileInfoList() throws StateStoreException, IOException, InterruptedException, IteratorException {
+        ingestRecords.init();
+        while (recordsIterator.hasNext()) {
+            ingestRecords.write(recordsIterator.next());
+        }
+        return ingestRecords.closeReturningFileInfoList();
+    }
```

Comment:
This duplicates the method above, ideally we would just return an IngestResult object.